### ISSUE TITLE
Comments

### ISSUE THREAD

**Issue by will-madebymoon** [dudelmeister/ONS-Sketch-Library#1]
Buttons should probably be laid out in the following way 

<img width="477" alt="screen shot 2019-02-08 at 14 15 20" src="https://user-images.githubusercontent.com/8184622/52483512-0b920900-2bac-11e9-8724-41b2b970997a.png">

**Comment by will-madebymoon** [comment]
I think we should probably review icon usage  and remove any that aren't used
<img width="293" alt="screen shot 2019-02-08 at 14 17 31" src="https://user-images.githubusercontent.com/8184622/52483653-5e6bc080-2bac-11e9-8384-a229ce273ddd.png">

**Comment by will-madebymoon** [comment]
We currently only allow the question H1 to be highlighted so no need for all of the variants. 
<img width="436" alt="screen shot 2019-02-08 at 14 20 10" src="https://user-images.githubusercontent.com/8184622/52483889-dd60f900-2bac-11e9-9d24-1944e4efaa71.png">

**Comment by will-madebymoon** [comment]
This is being deprecated so we should probably remove it
<img width="879" alt="screen shot 2019-02-08 at 14 24 10" src="https://user-images.githubusercontent.com/8184622/52484045-48123480-2bad-11e9-9802-8cbf4950c03c.png">

**Comment by will-madebymoon** [comment]
I think the structure would help being tidied up here a little so it's something like :

Components

- Buttons
       - Primary
       - Secondary
       - header
- Forms
       - Radio/checkboxes
       - Inputs
       - Selects
- Tables
- Navigation
       - Tabs

I was struggling to find stuff!

<img width="677" alt="screen shot 2019-02-08 at 14 25 42" src="https://user-images.githubusercontent.com/8184622/52484159-8f98c080-2bad-11e9-92b8-70eb8138fbf4.png">

**Comment by dudelmeister** [comment]
> Buttons should probably be laid out in the following way
> 
> <img alt="screen shot 2019-02-08 at 14 15 20" width="477" src="https://user-images.githubusercontent.com/8184622/52483512-0b920900-2bac-11e9-8724-41b2b970997a.png">

This is how I would've done it if I had made all the button pre-sets. Instead I have a master button which is fully customisable using overrides... Are we saying I should also make templates? Once we start doing this I feel like it defeats the point of nested symbols. I'm happy to go either way, be interested in seeing what others think too.

**Comment by dudelmeister** [comment]
> We currently only allow the question H1 to be highlighted so no need for all of the variants.
> <img alt="screen shot 2019-02-08 at 14 20 10" width="436" src="https://user-images.githubusercontent.com/8184622/52483889-dd60f900-2bac-11e9-9d24-1944e4efaa71.png">

I'm sure I've seen some screenshots where Rhod has used them on more than just the h1. I'll find out from him, but in the mean time - they are all there if needed

**Comment by dudelmeister** [comment]
> I think we should probably review icon usage and remove any that aren't used
> <img alt="screen shot 2019-02-08 at 14 17 31" width="293" src="https://user-images.githubusercontent.com/8184622/52483653-5e6bc080-2bac-11e9-8384-a229ce273ddd.png">

Agreed. We have a few different coloured chevrons and the direction of some chevrons differs dependent on the component's state, which is OK, but perhaps it's another consistency clean up task.

**Comment by dudelmeister** [comment]
> I think the structure would help being tidied up here a little so it's something like :
> 
> Components
> 
> * Buttons
>   - Primary
>   - Secondary
>   - header
> * Forms
>   - Radio/checkboxes
>   - Inputs
>   - Selects
> * Tables
> * Navigation
>   - Tabs
> 
> I was struggling to find stuff!
> 
> <img alt="screen shot 2019-02-08 at 14 25 42" width="677" src="https://user-images.githubusercontent.com/8184622/52484159-8f98c080-2bad-11e9-92b8-70eb8138fbf4.png">

Happy to make this change if we feel it's needed. My rationale for having it structured like this was around visibility. Nesting components in groups may make it harder to find for some.

**Comment by RhodGriffith** [comment]
> > We currently only allow the question H1 to be highlighted so no need for all of the variants.
> > <img alt="screen shot 2019-02-08 at 14 20 10" width="436" src="https://user-images.githubusercontent.com/8184622/52483889-dd60f900-2bac-11e9-9d24-1944e4efaa71.png">
> 
> I'm sure I've seen some screenshots where Rhod has used them on more than just the h1. I'll find out from him, but in the mean time - they are all there if needed

I think it's useful to keep these. I used it to highlight search results (h2) on the Business Register.

**Comment by RhodGriffith** [comment]
> > I think the structure would help being tidied up here a little so it's something like :
> > Components
> > 
> > * Buttons
> >   
> >   * Primary
> >   * Secondary
> >   * header
> > * Forms
> >   
> >   * Radio/checkboxes
> >   * Inputs
> >   * Selects
> > * Tables
> > * Navigation
> >   
> >   * Tabs
> > 
> > I was struggling to find stuff!
> > <img alt="screen shot 2019-02-08 at 14 25 42" width="677" src="https://user-images.githubusercontent.com/8184622/52484159-8f98c080-2bad-11e9-92b8-70eb8138fbf4.png">
> 
> Happy to make this change if we feel it's needed. My rationale for having it structured like this was around visibility. Nesting components in groups may make it harder to find for some.

Makes sense for the structure to reflect what we have in the new design system. I think these components were nested within 'form elements' on the old pattern library, but they're not in v2.

**Comment by RhodGriffith** [comment]
> > I think we should probably review icon usage and remove any that aren't used
> > <img alt="screen shot 2019-02-08 at 14 17 31" width="293" src="https://user-images.githubusercontent.com/8184622/52483653-5e6bc080-2bac-11e9-8384-a229ce273ddd.png">
> 
> Agreed. We have a few different coloured chevrons and the direction of some chevrons differs dependent on the component's state, which is OK, but perhaps it's another consistency clean up task.

There's also an edit icon in use on RM (internal) and a padlock icon used on RH social with the confidentiality statement.

**Comment by dudelmeister** [comment]
Had an interesting conversation with Dave House, one of the Designers assigned to the Design System. I asked him about folder structure and what the decision process was for not housing the components within sub headings. 

This was his response:

So there was 2 or 3 reasons we didn’t go with this…
1) Users interpretation of what group something would be in varied dramatically. We did multipl card sorts and testing and although say 50% of things were common there we so many things that weren’t.

2) The length of the groups mattered, for example if you started with an alphabetical list within “Form elements” and the list went off the screen users that weren’t looking for form elements just looked at the alphabetical list and thought the thing wasn’t on the page. As in I’m looking for Accordions, I can see that this is alphabetical therefore it’s not there.

3) There wasn’t a way of cutting the content in a way that there wasn’t always an other / misc section

I think we can take away some of this when we look at our own Design system (which we seem to be already) and extend some of the same logic to the folder structure in the Sketch library (especially point 1).

**Comment by will-madebymoon** [comment]
I’m cool with this however I made the observation from that fact I started using it and couldn’t find what I was after immediately (possible workflow issue). One of my biggest criticisms of the previous pattern library was that it was a component dump, it had very little structure and so for people who were new to EQ it would be very difficult to comprehend what it was. I also find V2 the component section a bit like that. 

As we develop new components we will also compound this issue and so it’s down to the ‘paradox of choice’, too much I don’t know what to do and perhaps too little and there is not enough here for me to do my job. I guess it’s like a carpenter or any kind of craftsman he arranges his tools best suited to his job. So with this in mind we have to ask what is purpose of the sketch library? are we right to align it completely to the design system? maybe/maybe not different users etc etc. 

Any how I’m more than happy to go with the general flow.

**Comment by will-madebymoon** [comment]
The only reason I mention this is because within runner and the author we are pretty strict what can be highlighted. 
If we allowed carte blanche usage then we’d see everything highlighted (Trust me! survey owners have attempted this.). 

If it’s only H1 or h2s then perhaps it should only be these that are highlighted, this would then prevent this from being over used. Our rationale is that only one thing per page should really be highlighted. 

Whilst we are initially the only users of this library there have been requests from other parts of the business to use our sketch files.
